src/js/views/patients/schedule/schedule_views.js

```diff
@@ -230,17 +277,44 @@ const ScheduleListView = CollectionView.extend({
       state: this.state,
     };
   },
-  emptyView: EmptyView,
+  childViewEvents: {
+    'filter'() {
+      this.debouncedChildFilter();
+    },
+  },
+  emptyView() {
+    if (this.state.get('searchQuery')) {
+      return EmptyFindInListView;
+    }
+
+    return EmptyView;
+  },
   viewComparator(viewA, viewB) {
     return alphaSort('asc', viewA.model.get('date'), viewB.model.get('date'));
   },
+  viewFilter(view) {
+    if (this.state.get('searchQuery')) {
+      return !view.children.isEmpty();
```

Comment:
I think just `view.isEmpty()` is ideal.  This allows the collectionview to decide when it is empty, if for some reason we inject some other view in that collectionview.

Reply:
We can't call `view.isEmpty()` because that view is never empty since the action item children are rendered into a container within that view. So we need to check the children.

However, there seems to be an order of operations issue here. When the list first renders `viewFilter` is called before any of the children are rendered. So `view.children.isEmpty()` is always true and the list renders empty. Which is the reason for the `searchQuery` check.